edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -913,12 +922,36 @@ export default {
         }
       }
 
+      if (!this.value.metadata.name && this.agentConfig['cloud-provider-name'] === HARVESTER) {
+        this.errors.push(this.t('validation.required', { key: this.t('cluster.name.label') }, true));
+      }
+
       if (this.errors.length) {
         btnCb(false);
 
         return;
       }
 
+      if (this.agentConfig['cloud-provider-name'] === HARVESTER) {
+        const clusterId = get(this.credential, 'decodedData.clusterId') || '';
+        const setting = await this.$store.dispatch('cluster/request', { url: `/k8s/clusters/${ clusterId }/v1/${ MANAGEMENT.SETTING }s/server-url` });
+        const serverUrl = (setting?.value || '').replace(/:[0-9]+/, '');
+        const namespace = this.machinePools?.[0]?.config?.vmNamespace;
+
+        const res = await this.$store.dispatch('management/request', {
+          url:                  `/k8s/clusters/${ clusterId }/v1/harvester/kubeconfig`,
+          method:               'POST',
+          data:                 {
+            serverUrl:          `${ serverUrl }:6443`,
+            clusterRoleName:    'harvesterhci.io:cloudprovider',
+            namespace,
+            serviceAccountName: this.value.metadata.name,
```

Comment:
Is something validating that this service account name doesn't already exist and user asking for this closet should have access to use it?  The name having to match seems suspicious.

Reply:
The backend will validate the service account name. 

UI can pass any name to the backend only if the name is not null. If the service account already exists, the backend will create a new one. If not, the backend will use the existing service account.

So I add cluster name validation to make sure the service account name is not null.

Reply:
@yasker or @n313893254 do you know if this validation was added yet or not?

Reply:
Will need some modification on this part since we also need to clean up the related SA once the cluster is removed by the user.